### PR TITLE
Added the ability to resize the sidebar

### PR DESCRIPTION
**Pull request by soumyamahunt**
This pr tries to implement feature requested in #672.

### CODE REVIEW

**Comment by soumyamahunt** [comment]
Currently sidebar can only be resized up to certain width.

**Comment by yaichenbaum** [comment]
> Currently sidebar can only be resized up to certain width.

@soumyamahunt That would be the expected behavior. Does the app keep track of the size across sessions?

**Comment by soumyamahunt** [comment]
> > Currently sidebar can only be resized up to certain width.
> 
> @soumyamahunt That would be the expected behavior. Does the app keep track of the size across sessions?

No not now, I will add it.

**Comment by duke7553** [comment]
@soumyamahunt If you need any inspiration, check out the appropriate branch for my past, unfinished work on this. Looks really good so far.

**Comment by soumyamahunt** [comment]
@yair100 @duke7553 what should we keep minimum and maximum sidebar width??

**Comment by yair100** [comment]
> @yair100 @duke7553 what should we keep minimum and maximum sidebar width??

I would recommend keeping the current size as the minimum.

**Comment by jaigak** [comment]
@yair100 To match Fluent Design Guidelines, the minimum width must be at least 20% of the container.

**Comment by Jaiganeshkumaran** [comment]
@soumyamahunt Could you share a screenshot on the changes?

**Comment by soumyamahunt** [comment]
> @soumyamahunt Could you share a screenshot on the changes?

Here is how it behaves with a minimum and maximum value.
R7Mjnsy4Dx

**Comment by soumyamahunt** [comment]
@yair100, @duke7553 minimum width is kept at 200(as was before) maximum is kept at 500. Also, width is remembered across instances.

R7Mjnsy4Dx

**Comment by Jaiganeshkumaran** [comment]
Can you increase the minimum width to match the default WinUI styles?

**Comment by soumyamahunt** [comment]
> Can you increase the minimum width to match the default WinUI styles?

I can but @yair100 suggested to keep it same as before, @yair100 what do you think of this?? Also what is the default width value in WinUI style??

**Comment by duke7553** [comment]
@soumyamahunt Nope, we're happy with the existing minimum.

File: Files/Strings/de-DE/Resources.resw [hunk 1]
```diff
@@ -666,4 +723,43 @@
...
+  </data>
+  <data name="SettingsOnStartupLaunchNewTab.Content" xml:space="preserve">
+    <value>Ordner in neuem Tab öffnen</value>
+  </data>
+  <data name="SettingsPreferencesSystemDefaultLanguageOption" xml:space="preserve">
+    <value>Windows standard</value>
+  </data>
+  <data name="TabStripDragAndDropUIOverrideCaption" xml:space="preserve">
+    <value>Tab hierhin verschieben</value>
+  </data>
```
Comment (yaichenbaum) [inline Files/Strings/de-DE/Resources.resw:764]: Please revert the changes to the strings.

File: Files/Views/ModernShellPage.xaml.cs [hunk 2]
```diff
@@ -26,6 +26,26 @@ public sealed partial class ModernShellPage : Page, IShellPage
...
+                }
+
+                return new GridLength(sidebarWidth, GridUnitType.Pixel);
+            }
+            set
+            {
+                ApplicationData.Current.LocalSettings.Values["SidebarWidth"] = value.Value;
+            }
+        }
+
```
Comment (yaichenbaum) [inline Files/Views/ModernShellPage.xaml.cs:48]: Please use this format for saving the settings.
```
public bool OpenASpecificPageOnStartup
{
        get => Get(false);
        set => Set(value);
}
```

**Comment by Jaiganeshkumaran** [comment]
> @soumyamahunt Nope, we're happy with the existing minimum.

Now the settings sidebar looks out of place. Could you use the same size for both NavigationView controls and also make Setting sidebar resizable?

File: Files/Views/ModernShellPage.xaml.cs (hunk 2 above)
Comment (soumyamahunt) [reply Files/Views/ModernShellPage.xaml.cs]: Made changes.

**Comment by soumyamahunt** [comment]
> Now the settings sidebar looks out of place. Could you use the same size for both NavigationView controls and also make Setting sidebar resizable?

I don't think making settings sidebar resizable gonna improve any usability since the strings used there are static, but I can make it sync with the main sidebar. @yair100, is this something you would like too??

**Comment by yaichenbaum** [comment]
The settings sidebar should not be resizable.

**Review by yair100 — APPROVED** [review]
